lib/connection-manager.js

```diff
@@ -65,21 +27,44 @@ module.exports.connectLogServer = function(config, programName) {
     };
     var proxy = getProxyFromURI(url.parse(logurl));
     if (proxy) options.agent = new HttpsProxyAgent(url.parse(proxy));
-
-    var firstrun = true;
     debug(logurl);
-    var socket = new WebSocket(logurl, options);
-    linebuffer.setSocket(socket);
-    debug(socket.options);
-    return new Promise(resolve => {
-        debug('connecting to websocket');
-        socket.on('open', function() {
-            log('Connected to ' + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT + ' (' + socket._socket.remoteAddress + ')' + (config.LOGDNA_LOGSSL ? ' (SSL)' : '') + (proxy ? ' via ' + proxy : ''));
-
-            if (firstrun) {
-                firstrun = false;
-
-                // start streaming logs from logdir(s) on startup
+    const socket = new WebSocket(logurl, options);
+    const req = linebuffer.setConnection(logurl, options);
+    return new Promise((resolve) => {
+        async.retry({
+            times: 5
+            , interval: 100
+            , errorFilter: (err) => {
+                return [
+                    'ETIMEDOUT'
+                    , 'ECONNRESET'
+                    , 'EHOSTUNREACH'
+                    , 'ETIMEDOUT'
+                    , 'ESOCKETTIMEDOUT'
+                    , 'ECONNREFUSED'
+                    , 'ENOTFOUND'
+                ].indexOf(err.code) >= 0;
+            }
+        }, (reqCallback) => {
+            return req.post(`${logurl}${config.LOGDNA_LOGENDPOINT}`, {
+                body: JSON.stringify({
+                    e: 'ls'
+                    , ls: [{
+                        e: 'l'
+                        , l: `Connected to ${config.LOGDNA_LOGHOST}:${config.LOGDNA_LOGPORT} ${(config.LOGDNA_LOGSSL ? ' (SSL)' : '')}${(proxy ? ' via ' + proxy : '')}`
```

Comment:
seems like this statement is slightly different from the actual url being used (line:49), should these be the same to reflect what url you are actually making a post to? I mean assuming that it posted to the correct location to begin with.

Reply:
Yes, it does - removed backquotes to improve readability.